sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TransmitterFactory.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    /// <summary>
+    /// This Factory encapsulates the <see cref="AzureMonitorTransmitter"/>.
+    /// An ideal users will create a single exporter for each signal (Logs, Metrics, Traces).
+    /// This factory should ensure that only one instance of the Transmitter is created for
+    /// any unique connection string.
+    /// </summary>
+    internal class TransmitterFactory
+    {
+        public static TransmitterFactory Instance = new();
+
+        internal readonly Dictionary<string, AzureMonitorTransmitter> _transmitters = new();
```

Comment:
can we use ConcurrentDictionary instead?

Reply:
good question. 
this dictionary isn't going to see heavy use, so I'm not sure that we need the extra overhead.

Reply:
adding @rajkumar-rangaraj here. i'm not sure that ConcurrentDictionary avoids the extra lock but I'll get more information.

At worse, ConcurrentDictionary would prevent the second Add, but if we're calling Add a second time that means we've already created a second Transmitter :(

Reply:
can't we use something like dispose pattern here instead of locks?

Reply:
the current lock approach does not ever "waste" a Transmitter due to locks. It should be fine.

Reply:
Locks are primary cause for thread contention. If there are other ways we should avoid its usage.

Reply:
ConcurrentDictionary with a LazyInitializer might work. 
Though i'm worried that would be overengineering with a lot of extra overhead.

Reply:
this lock is not in the hot path, right? so should not be a concern

Reply:
I was referring to using this https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getoradd?view=net-7.0.

Reply:
GetOrAdd() is a good idea, but reading the description I don't think this meets our needs here.

> For modifications and write operations to the dictionary, [ConcurrentDictionary<TKey,TValue>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-7.0) uses fine-grained locking to ensure thread safety. (Read operations on the dictionary are performed in a lock-free manner.) However, the valueFactory delegate is called outside the locks to avoid the problems that can arise from executing unknown code under a lock. Since a key/value can be inserted by another thread while valueFactory is generating a value, you cannot trust that just because valueFactory executed, its produced value will be inserted into the dictionary and returned.